### DISCUSSION TITLE
Converting lines of text/word file to separate PNG files

### DISCUSSION THREAD

**Discussion by webbooo** [ImageMagick/ImageMagick#5025] [Help] [answered]
Hi Everyone,

I am new here (excuse my ignorance) as only just discovered this tool through a google search.

I have a file with individual lines of text.  I'm trying to see if there is a way to convert each line item into a separate png file, where each png is 300 x 300 pixels and the text should be positioned in the bottom right hand corner (Text can have a hight of 40 pixels).  Not sure if this is possible, if anyone has any suggestion or somewhere to point me Id appreciate your help.

Thanks
Adam

**Comment by snibgo** [comment]
What version of IM, on what platform?

You could have a shell loop that reads each line of text. For each line, it executes an IM command. That command would create a PNG file from the text.

**Reply by webbooo** [reply]
Hi,  version is 7.1 and I'm on win 10.

Thanks

**Reply by fmw42** [reply]
Post an example image.  You could use morphology to make each line of text a block of black, then use connected components to extract each block.

EDIT:  Sorry, I thought you had an image of text, not an ascii or word text file.

**Reply by webbooo** [reply]
no  worries, I didnt make that overly clear but yea just starting with a txt file

**Comment by snibgo** [comment]
Suppose sample.txt contains this:
```
first line
second line
third line
```

Then you could have a Windows BAT script:
```
set NUM=0

for /F "tokens=*" %%L in (sample.txt) do (
  %IMG7%magick -size 300x300 -gravity SouthEast -pointsize 40 caption:"%%L" x_!NUM!.png
  set /A NUM+=1
)
```
That creates files x_0.png, x_1.png and x_2.png.

**Reply by webbooo** [reply]
thanks, this somewhat worked but only produced one png which was of the last line item in the text file, so in this case 'third line'.  I made some changes to the file and it didn't matter how many line items it only produced a single png of last line.  Script appeared to look at each line but didnt produce a png for top two.

**Reply by webbooo** [reply]
the name of the output file was also 'x_!NUM!.png' and not x_0.png etc

**Reply by webbooo** [reply]
would there be a way to make the output file name the same as the text?

**Comment by snibgo** [comment, marked as answer]
For the "!" problem, insert a new line "setlocal enabledelayedexpansion" at the top of the script.

In the script, "%%L" is the text line. If you want the files to be named "first line.png" etc, then use "%%L.png", including the quotes, as the output filename.

**Reply by webbooo** [reply]
thank you, this worked perfectly.

appreciate you taking them time to help.
